### PR TITLE
Add additional tests for unexpected fields in units

### PR DESCRIPTION
**Pull request by kroenlein**
Field testing showed some bad behavior:
* Punctuation like `"` and `:` were being swallowed, leading to things that weren't valid units being accepted
* Strings that were just numbers being being interpreted as valid units

### CODE REVIEW

File: gemd/units/tests/test_parser.py [hunk 1]
```diff
@@ -42,12 +42,17 @@ def test_parse_unexpected():
         5,
         "cp",  # Removed because of risk of collision with cP
         "chain",  # Survey units eliminated
-        "SECONDS"  # Not just case insensitivity
+        "SECONDS",  # Not just case insensitivity
+        "lb : in^3",  # Not just case insensitivity
```
Comment (bfolie) [inline gemd/units/tests/test_parser.py:46]: This comment seems unrelated

Comment (kroenlein) [reply gemd/units/tests/test_parser.py:46]: Good eyes, copypasta

File: gemd/units/impl.py [hunk 2]
```diff
@@ -26,6 +26,9 @@ def _scaling_preprocessor(input_string: str) -> str:
     tight_division = False
     scales = []
 
+    if next(token for token in tokens).type == NUMBER:
+        return input_string  # The unit can't have a leading number; scaling factors are internal
```
Comment (bfolie) [inline gemd/units/impl.py:30]: Why does returning the input string lead to the desired error being thrown?

Also, why is a leading number not allowed? If `g / 100 L` is then why not `100 L / g`? Or is this to cover for the cases in which the input string is _just_ a number?

File: gemd/units/impl.py [hunk 3]
```diff
@@ -36,8 +39,13 @@ def _scaling_preprocessor(input_string: str) -> str:
         if not exponent and token.type == NUMBER:
             scales.append([token.string, False])
             tight_division = division
-        exponent = token.type == OP and token.string in {"^", "**"}
-        division = token.type == OP and token.string in {"/", "//"}
+        if token.type == OP:
+            if token.string not in {"+", "-", "*", "/", "//", "^", "**", "(", ")"}:
```
Comment (bfolie) [inline gemd/units/impl.py:43]: This would be more readable if the set of all allowed tokens was broken out as a constant

File: gemd/units/impl.py (hunk 2 above)
Comment (kroenlein) [reply gemd/units/impl.py:30]: We exit the preprocessor without mangling the leading digit, so it's handled as it would have been handled by default.  We could probably get equivalent functionality by just swallowing the first term, but I didn't want to have to deal with possible weirdness following from that if that first term would have set some of the internal booleans.

No leading digit because normal Pint behavior says that should be a magnitude on a value, and that kinda matches.  I personally also can't think of a customary unit that has a scaling factor that has it up front like that.  It conveniently also covers the case where it's just a number.

File: gemd/units/impl.py (hunk 3 above)
Comment (kroenlein) [reply gemd/units/impl.py:43]: Fair.  I'd wanted to import it from Pint, but it's a private variable.

**Review by bfolie — APPROVED** [review]
I don't fully understand it, but the tests pass and they look appropriate

**Review by bfolie — APPROVED** [review]
